### PR TITLE
Refactor to use ember-modifier's modify hook and fix deprecation

### LINKED ISSUE #452: Triggers deprecations with ember-modifier 3.2.0+
ember-modifier 3.2.0 introduced a new `modify` hook, and added a number of deprecations: https://github.com/ember-modifier/ember-modifier/blob/master/addon/CHANGELOG.md#v320-2022-03-29

These are all triggered by this modifier now.

### PR DESCRIPTION
**Pull request by st-h**
fixes #452

### CODE REVIEW

**Comment by bwbuchanan** [comment]
Hi @alexlafroscia - is this addon still maintained?

It would be great if this PR could be merged and released to resolve the `ember-modifier` deprecation warnings.

**Comment by bwbuchanan** [comment]
@st-h thanks for that. I actually don't depend on this addon directly, but via ember-bootstrap. Hopefully ember-bootstrap can make the switch to a maintained addon.

**Comment by jelhan** [comment]
@alexlafroscia Ember Bootstrap depends on this. I would love landing support for `ember-modifier@^4` in Ember Bootstrap soon. This requires an `ember-popper-modifier` version, which supports `ember-modifier@^4` itself. First step would be migrating to `modify` hook as that one is supported by both `ember-modifier@^3.2` and `ember-modifier@^4`. I can support you on landing all that changes.

File: addon/-base-popper-modifier.js [hunk 1]
```diff
@@ -65,7 +66,12 @@ export default class PopperModifier extends Modifier {
     return options;
   }
 
-  didReceiveArguments() {
+  modify(element, positionalArgs, namedArgs) {
+    this.primaryElement = element;
+    this.secondaryElement = positionalArgs[0];
+
+    this.popperOptions = this.initPopperOptions(positionalArgs, namedArgs);
```
Comment (jelhan) [inline addon/-base-popper-modifier.js:73]: Why is this set on the class instance? It seems to be only used in this method. It seems as if we could refactor to a simple variable. Or did I missed something?

```suggestion
    const popperOptions = this.initPopperOptions(positionalArgs, namedArgs);
```

**Review by jelhan — COMMENTED** [review]
Thanks a lot for starting the work on this one! I left two comments for improving readability of the code. Do you have time to address them?

**Comment by st-h** [comment]
@jelhan sorry, I no longer use this addon. Please feel free to make any needed adjustments to this PR.

File: addon/-base-popper-modifier.js (hunk 1 above)
Comment (st-h) [reply addon/-base-popper-modifier.js:73]: I pretty much just tried to not change too many things, so I kept the style of making options available on the instance.

**Pull request merged**